ImperatorToCK3/Source/Mappers/CultureMapper/CultureMappingRule.cpp

```diff
@@ -56,17 +60,56 @@ std::optional<std::string> mappers::CultureMappingRule::cultureMatch(const std::
 			return std::nullopt;
 	}
 
-	// This is a straight province check
-	if (CK3Province && !provinces.empty())
-		if (!provinces.contains(CK3Province))
+	if (!ck3Provinces.empty() || !imperatorProvinces.empty() || !ck3Regions.empty() || !imperatorRegions.empty())
```

Comment:
While you're touching this, it's probably simpler to negate the conditional and have the complex code at one fewer level of nesting:

```
// simple culture-culture match
if (ck3Provinces.empty() && imperatorProvinces.empty() && ck3Regions.empty() & imperatorRegions.empty())
{
	return destinationCulture;
}
```

Reply:
Thanks, did the same for ReligionMapping.